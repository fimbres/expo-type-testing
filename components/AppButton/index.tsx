import React, { FC } from "react";
import { Pressable, StyleSheet, Text } from "react-native";

import { Colors, Styles } from "../../constants";

interface AppButtonProps {
  title: string;
  size?: "md" | "lg";
  onPress: () => void;
}

export const AppButton: FC<AppButtonProps> = ({
  title,
  size = "md",
  onPress,
}) => {
  return (
    <Pressable
      style={({ pressed }) => [
        styles.container,
        styles[size],
        pressed && styles.pressed,
      ]}
      onPress={onPress}
      testID="app-button"
    >
      <Text
        style={[
          size === "md" ? Styles.textTitleExtraSmall : Styles.textTitleRegular,
          { color: Colors.white },
        ]}
      >
        {title}
      </Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center",
    borderRadius: 10,
    backgroundColor: Colors.blue,
  },
  md: {
    paddingVertical: 17,
  },
  lg: {
    paddingVertical: 14,
  },
  pressed: {
    opacity: 0.8,
  },
});
