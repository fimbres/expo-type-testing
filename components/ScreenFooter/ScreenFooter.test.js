import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react-native";

import { ScreenFooter } from ".";
import { AppButton } from "../AppButton";

describe("<ScreenFooter />", () => {
  it("render children correctly", () => {
    render(
      <ScreenFooter>
        <AppButton title="Test One" onPress={console.log} />
        <AppButton title="Test Two" onPress={console.log} />
      </ScreenFooter>
    );

    const firstText = screen.getByText("Test One");
    const secondText = screen.getByText("Test Two");

    expect(firstText).toBeDefined();
    expect(secondText).toBeDefined();
  });

  it("render correctly", () => {
    const tree = renderer
      .create(
        <ScreenFooter>
          <AppButton title="Test One" onPress={console.log} />
          <AppButton title="Test Two" onPress={console.log} />
        </ScreenFooter>
      )
      .toJSON();
    expect(tree.children.length).toBe(2);
    expect(tree).toMatchSnapshot();
  });
});
