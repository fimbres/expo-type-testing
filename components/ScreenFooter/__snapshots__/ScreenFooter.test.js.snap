// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenFooter /> render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginVertical": 40,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#334FFA",
          "borderRadius": 10,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "paddingVertical": 17,
        },
        false,
      ]
    }
    testID="app-button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#020202",
            "fontFamily": "avenir-black",
            "fontSize": 12,
            "fontWeight": "800",
            "lineHeight": 16.39,
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Test One
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#334FFA",
          "borderRadius": 10,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "paddingVertical": 17,
        },
        false,
      ]
    }
    testID="app-button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#020202",
            "fontFamily": "avenir-black",
            "fontSize": 12,
            "fontWeight": "800",
            "lineHeight": 16.39,
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Test Two
    </Text>
  </View>
</View>
`;
